config/config.go

```
@@ -63,21 +64,143 @@ type DataStore struct {
 	DBNumber int
 }
 
+var defaultConfig = &config{
+	Munch: Munch{
+		Server: MunchServer{
+			Hostname: "localhost",
+			Port:     "50050",
+		},
+		LogLevel: "info",
+	},
+	DocumentStore: "mongo",
+	Mongo: DataStore{
+		Name:     "mongo",
+		URI:      "mongodb://localhost:27017",
+		DBName:   "meteomunch",
+		DBNumber: 0,
+	},
+	DLMRedis: DataStore{
+		Name:     "redis",
+		URI:      "redis://localhost:6379",
+		DBNumber: 1,
+	},
+	MeteoProviders: []MeteoProvider{
+		{
+			Name:    "open-meteo",
+			APIKey:  "",
+			APIPath: "v1/forecast",
+			BaseURI: "https://api.open-meteo.com/",
+		},
+	},
+}
+
 var legal = &struct {
 	documentStore []string
 }{
 	documentStore: []string{"mongo"},
 }
 
-// TODO: This can be more perfomant by loading it just once and pass a copy for each call
+// New creates a new config object
 func New() (*config, error) {
-	// log := logger.NewTag("config")
 	conf := new(config)
 
+	pwd, err := os.Getwd()
+	cobra.CheckErr(err)
+	viper.AddConfigPath(pwd)
+	viper.AddConfigPath(".")
+	viper.SetConfigType("yml")
+	viper.SetConfigName("munch")
+
+	if err := viper.ReadInConfig(); err != nil {
+		slog.Warn("Config file not found, using default config", "error", err)
+		return defaultConfig, nil 
+	}
+
 	if err := viper.Unmarshal(conf); err != nil {
-		slog.Error(e.FAIL, "err", err, "description", "Couldn't parse config")
-		return nil, err
+		slog.Error(e.FAIL, "err", err, "description", "Couldn't parse config, using default config")
+		return defaultConfig, nil 
 	}
 
+	setDefaultValues(conf)
+
+	// Check for critical values and panic if missing
+	validateCriticalFields(conf)
```

Ideally, this must be run prior to the setting of default values so that it fails fast.

---

Should this fail, `setDefaultValues()` need not run.

---

Okay